pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -542,3 +557,59 @@ func implicitWaypointPolicies(ctx krt.HandlerContext, Waypoints krt.Collection[W
 		return ptr.Of(w.Namespace + "/" + policy)
 	})
 }
+
+func gatewayUID(gw model.NetworkGateway) string {
+	return "NetworkGateway/" + string(gw.Network) + "/" + gw.Addr + "/" + strconv.Itoa(int(gw.HBONEPort))
+}
+
+// convertGateway always converts a NetworkGateway into a Workload
+func convertGateway(gw model.NetworkGateway) model.WorkloadInfo {
```

Comment:
Why should convert to workload?

Reply:
workload it self has NetworkGateway fields can be used

Reply:
NetworkGateway field in workload is effectively a pointer to the gateway. it does not have sufficient information to call the gateway one its own. The same applies to waypoint today. So this gives the additional information we need.

Ill improve the comments in the code

Reply:
Yeah, i see SA is needed, but how about expanding the Workload api's GatewayAddress to facilitate lookup

Reply:
the intent of the API is to make it a pointer so we can encode as much information as we need there. For instance, what if it's a service backed by 10 pods, each with a service account, Telemetry information, load balancing policies, etc.